regulations/views/preamble.py

```diff
@@ -60,6 +61,12 @@ def generate_html_tree(subtree, request, id_prefix=None):
             'markup_page_type': 'reg-section'}
 
 
+def get_toc_position(toc, ids):
+    return next((index for index, value in enumerate(toc[0].sections)
+                if value.part == ids['part']
+                and value.section == ids['section']), None)
```

Comment:
I led you a bit astray when we spoke before; we'll want to iterate over the `toc` list (it has one entry per CFR part).

Minor stuff: would a for loop be a little easier to read? Similarly, do we want named args? e.g.
```python
def get_toc_position(toc, part, section):
    sections = [section for part in toc for section in part.sections]
    for index, value in enumerate(sections):
        if value.part == part and value.section == section:
            return index
```

```python
toc_position = get_toc_position(context['cfr_change_toc'], **ids)
```